scico/linop/_diag.py

```diff
@@ -156,10 +156,137 @@ def norm(self, ord=None):  # pylint: disable=W0622
             mord = snp.inf
         if mord not in ordfunc:
             raise ValueError(f"Invalid value {ord} for parameter ord.")
-        return ordfunc[mord](self.diagonal)
+        return ordfunc[mord](self._diagonal)
+
+
+class ScaledIdentity(Diagonal):
+    """Scaled identity operator."""
+
+    def __init__(
+        self,
+        scalar: float,
+        input_shape: Union[Shape, BlockShape],
+        input_dtype: DType = snp.float32,
+        **kwargs,
+    ):
+        """
+        Args:
+            scalar: Scaling of the identity.
+            input_shape: Shape of input array.
+            input_dtype: `dtype` of input argument.
+        """
+        if is_nested(input_shape):
+            diagonal = scalar * snp.ones(((),) * len(input_shape), dtype=input_dtype)
```

Comment:
I assume you tried `diagonal = scalar`. I'm curious why that doesn't work.

Reply:
The initializer for `Diagonal` assumes that `_diagonal` has a `shape` attribute:
https://github.com/lanl/scico/blob/7cef2aa41d0f333e05c940add6ce89199a42a4ea/scico/linop/_diag.py#L56-L59
This could, of course, be modified to check for the scalar case, but that would just entail moving this complexity elsewhere in the code.